cpp/include/ucxx/endpoint.h

```diff
@@ -418,6 +419,7 @@ class Endpoint : public Component {
   std::shared_ptr<Request> tagRecv(void* buffer,
                                    size_t length,
                                    ucp_tag_t tag,
+                                   ucp_tag_t tagMask,
```

Comment:
This is an excellent suggestion. I've been meaning to work on something like that, starting with the [`bool send`](https://github.com/rapidsai/ucxx/blob/branch-0.36/cpp/src/request_tag.cpp#L17) we have in requests which is probably the worst such case in UCXX currently, but tags should do the same.

This is now done via https://github.com/rapidsai/ucxx/pull/121/commits/d706c3299aa2160e62c00d139a395399a7a28e0c, as a bonus I attempted to do the same on the Python side via https://github.com/rapidsai/ucxx/pull/121/commits/9dac813cf06429772a89cb7cfc0c19a6de839d3a. I'm not confident this is the best for Python yet, WDYT?

Reply:
I went ahed and changed the `bool send` cases to the new `TransferDirection` as well via https://github.com/rapidsai/ucxx/pull/121/commits/89e17a6b6885d1461207a185e9083f7bf876b676, https://github.com/rapidsai/ucxx/pull/121/commits/3d50c74142f42aa3056ece6d7a41ce5bf2767de5 and https://github.com/rapidsai/ucxx/pull/121/commits/c8c656f61ed50dd0a99cd9d308105ac1da4e7643. I'm gonna stop here though otherwise I'll attempt to fix all the weak typing in this one PR alone.